src/controllers/qasm_controller.hpp

```diff
@@ -962,16 +962,19 @@ void QasmController::run_multi_shot(const Circuit& circ,
     // Run circuit instructions before first measure
     std::vector<Operations::Op> ops(circ.ops.begin(),
                                     circ.ops.begin() + pos);
+    bool final_ops = (pos == circ.ops.size());
     initialize_state(circ, state, initial_state);
-    state.apply_ops(ops, data, rng);
+    state.apply_ops(ops, data, rng, final_ops);
 
-    // Get measurement operations and set of measured qubits
-    ops = std::vector<Operations::Op>(circ.ops.begin() + pos,
-                                      circ.ops.end());
-    measure_sampler(ops, shots, state, data, rng);
+    if (!final_ops) {
```

Comment:
Is this needed? If I remove this all tests pass.

Reply:
Its not needed, the functions in that block shouldn't do anything if `final_ops = false` anyway